File: languages/go/entity/identity/publish_interval_params.go
package identity

import "github.com/bloock/bloock-sdk-go/v2/internal/bridge/proto"

// PublishIntervalParams represents different publish intervals.
type PublishIntervalParams int32

const (
	// Interval5 represents a 5-minute publish interval.
	Interval5 PublishIntervalParams = iota
	// Interval15 represents a 15-minute publish interval.
	Interval15 PublishIntervalParams = iota
	// Interval60 represents a 60-minute publish interval.
	Interval60 PublishIntervalParams = iota
)

var (
	PublishIntervalParamsFromProto = map[proto.PublishInterval]PublishIntervalParams{
		proto.PublishInterval_INTERVAL_5:  Interval5,
		proto.PublishInterval_INTERVAL_15: Interval15,
		proto.PublishInterval_INTERVAL_60: Interval60,
	}

	PublishIntervalParamsToProto = map[PublishIntervalParams]proto.PublishInterval{
		Interval5:  proto.PublishInterval_INTERVAL_5,
		Interval15: proto.PublishInterval_INTERVAL_15,
		Interval60: proto.PublishInterval_INTERVAL_60,
	}
)

File: languages/js/src/entity/identity/publish_interval_params.ts
import { PublishInterval } from "../../bridge/proto/bloock_identity_entities";

/**
 * Represents different publish intervals.
 */
export enum PublishIntervalParams {
  /**
   * Represents a 5-minute publish interval.
   */
  Interval5,
  /**
   * Represents a 15-minute publish interval.
   */
  Interval15,
  /**
   * Represents a 60-minute publish interval.
   */
  Interval60
}

export namespace PublishIntervalParams {
  export function toProto(type: PublishIntervalParams): PublishInterval {
    switch (type) {
      case PublishIntervalParams.Interval5:
        return PublishInterval.INTERVAL_5;
      case PublishIntervalParams.Interval15:
        return PublishInterval.INTERVAL_15;
      case PublishIntervalParams.Interval60:
        return PublishInterval.INTERVAL_60;
      default:
        return PublishInterval.INTERVAL_60;
    }
  }

  export function fromProto(
    type: PublishInterval | undefined
  ): PublishIntervalParams {
    switch (type) {
      case PublishInterval.INTERVAL_5:
        return PublishIntervalParams.Interval5;
      case PublishInterval.INTERVAL_15:
        return PublishIntervalParams.Interval15;
      case PublishInterval.INTERVAL_60:
        return PublishIntervalParams.Interval60;
      default:
        return PublishIntervalParams.Interval60;
    }
  }
}
